### PR TITLE
mkcloud, qa_crowbarsetup: Add support for SLE12 SP1 for Cloud 6

### PR DESCRIPTION
**Pull request by vuntz**
This is based on a new want_sles12sp1 option, that (hopefully) will be
dropped once the port to SP1 is fully completed and we won't use SLE12
GA.

### CODE REVIEW

**Comment by MaximilianMeister** [comment]
> Why hacloud == 2 ?

usually that was intended as a boolean. this is really misused, especially if it is not documented anywhere. if i would misuse it, i would misuse it correctly like `hacloud == disable_ha_testrepo`

**Comment by vuntz** [comment]
> usually that was intended as a boolean. this is really misused, especially if it is not documented anywhere.

I'm not disagreeing, but this is unrelated to this PR, really :-) Feel free to submit a PR to fix this (or create a card in the backlog).

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -610,9 +643,13 @@ function get_crowbar_node()
...
 }
 
-function get_sles12_node()
+function get_sles12plus_node()
 {
-    knife search node "target_platform:suse-12.0" -a name | grep ^name: | cut -d : -f 2 | sort | tail -n 1 | sed 's/\s//g'
+    if [ -n "$want_sles12sp1" ]; then
+        knife search node "target_platform:suse-12.1" -a name | grep ^name: | cut -d : -f 2 | sort | tail -n 1 | sed 's/\s//g'
+    else
+        knife search node "target_platform:suse-12.0" -a name | grep ^name: | cut -d : -f 2 | sort | tail -n 1 | sed 's/\s//g'
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: this is a bit ugly code duplication - maybe we could always look for suse-12.0 and 12.1 nodes... or make the number variable like
```bash
x=suse-12.0
[[ $want_sles12sp1 ]] && x=suse-12.1
```

Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: fixed

**Comment by toabctl** [comment]
+1

**Comment by dirkmueller** [comment]
+1

**Pull request merged**